src/test/java/org/opensearch/ad/util/LabelledAnomalyGenerator.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.util;
+
+import static java.lang.Math.PI;
+
+import java.util.Random;
+
+import org.joda.time.Instant;
+
+/**
+ * Regarding benchmark data, we randomly generated synthetic data with known anomalies
+ * inserted throughout the signal. In particular, these are one/two/four dimensional
+ * data where each dimension is a noisy cosine wave. Anomalies are inserted into one
+ * dimension with 0.003 probability. Anomalies across each dimension can be independent
+ * or dependent.  We have approximately 5000 observations per data set. The data set is
+ * generated using the same random seed so the result is comparable across versions.
+ *
+ */
+public class LabelledAnomalyGenerator {
```

Comment:
This is nice to have. I wonder if we could expose this on the UI somewhere to let users play around with some data and test out detectors for themselves.

Reply:
Also, what was the reasoning for only testing 1/2/4 dimensional data? Do higher dimensions bring too much variability/noise regarding precision/recall results?

Reply:
yeah, this could be part of HCAD sample detectors.

Reply:
By default, we support up to 5 features, even though we can increase to 10. Testing 1/2/4 dimensional data would cover power of 2 dimensions less than 5. Yeah, we can do all 5 of them. I assume 1/2/4 can give us a glimpse of the whole picture. Yes, high dimensions brings curse of dimensionality (https://en.wikipedia.org/wiki/Curse_of_dimensionality).

Reply:
When the number of features increase to 4, the precision/recall is lesser. Take 2.4 as an example:

| Precision      | Recall |  Number of features |
| ----------- | ----------- |
| 0.5697 | 0.4548  | 4 |
| 0.62214 | 0.64622| 2 |
| 0.62944 | 0.58078 | 1 |

There is only 1 unique entity involved. You can see I only have one entity in HCADModelPerfTests.averageAccuracyTemplate.

```
entity = Entity.createSingleAttributeEntity("field", entityName + z);
```